lib/SILOptimizer/SILCombiner/SILCombinerBuiltinVisitors.cpp

```diff
@@ -109,6 +109,49 @@ SILInstruction *SILCombiner::optimizeBuiltinIsConcrete(BuiltinInst *BI) {
   return Builder.createIntegerLiteral(BI->getLoc(), BI->getType(), 1);
 }
 
+/// Replace
+/// \code
+///   %b = builtin "COWBufferForReading" %r
+///   %a = ref_element_addr %b
+/// \endcode
+/// with
+/// \code
+///   %a = ref_element_addr [immutable] %r
+/// \endcode
+/// The same for ref_tail_addr.
+SILInstruction *SILCombiner::optimizeBuiltinCOWBufferForReading(BuiltinInst *BI) {
```

Comment:
hmmm... this seems like it could be an instruction simplification since you're not creating a new `ref_element_addr`

Reply:
But the builtin is deleted. AFAIK, it shouldn't be done in simplification.

Reply:
yeah, it's not the usual inst-simplify, mainly because the users of the ref_element_addr don't actually have their operand replaced, so the driver won't think anything changed